src/theory/datatypes/theory_datatypes_type_rules.h

```diff
@@ -427,7 +427,115 @@ class DtSyguEvalTypeRule
     }
     return TypeNode::fromType(dt.getSygusType());
   }
-}; /* class DtSygusBoundTypeRule */
+}; /* class DtSyguEvalTypeRule */
+
+class MatchTypeRule
+{
+ public:
+  inline static TypeNode computeType(NodeManager* nodeManager,
```

Comment:
Oh right, I missed that one. @4tXJ7f should we add something about when to (not) use `inline` (i.e., always, don't use it?) to the coding guidelines?

Reply:
Yeah, that may be a good idea. I think that the rule should be that we _never_ use `inline` since it's non-binding in terms of whether the code actually gets inlined (and we probably don't need multiple definitions of the same function). We should just let the compiler decide.

Reply:
@4tXJ7f can you add this to the code guidelines?

Reply:
Added to the code guidelines [here](https://github.com/CVC4/CVC4/wiki/Code-Style-Guidelines#inline-functions).